src/sw/redis++/connection.cpp

```diff
@@ -93,8 +93,6 @@ void ConnectionOptions::_set_option(const std::string &key,
         opts.connect_timeout = _parse_timeout_option(val);
     } else if (key == "socket_timeout") {
         opts.socket_timeout = _parse_timeout_option(val);
-    } else {
```

Comment:
We need to keep this check. In case that user has spelling error on options, e.g. *Sockttimeout*. If we don't throw exception, user would think s/he successfully set the parameter.

Reply:
How about an overloaded constructor for connectionoptions that accepts an optional boolean for `allow_unknown_uri_parameters`? That wouldn't change existing behavior.